src/ngraph/runtime/gpu/nvrtc/helpers.hpp

```diff
@@ -0,0 +1,208 @@
+//*****************************************************************************
+// Copyright 2017-2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#pragma once
+
+#include <cuda.h>
+#include <sstream>
+#include <string>
+
+#include "ngraph/except.hpp"
+
+namespace ngraph
+{
+    namespace runtime
+    {
+        namespace gpu
+        {
+            namespace nvrtc
+            {
+                std::string helpers();
+                std::string define_zero(const std::string& dtype,
+                                        const std::string& name = "zero_");
+                std::string define_vzero(std::string dtype,
+                                         const uint32_t& n,
+                                         const std::string& name = "zero_");
+                std::string define_coherent_load(const std::string& dtype,
+                                                 const std::string& name = "load_");
+                std::string define_coherent_vload(const std::string& dtype,
+                                                  const uint32_t& n,
+                                                  const std::string& name = "load_");
+                std::string define_non_coherent_load(const std::string& dtype,
+                                                     const std::string& name = "load_");
+                std::string define_non_coherent_vload(const std::string& dtype,
+                                                      const uint32_t& n,
+                                                      const std::string& name = "load_");
+            }
+        }
+    }
+}
+
+std::string ngraph::runtime::gpu::nvrtc::helpers()
+{
+    std::stringstream ss;
+#if defined(CUDA_VERSION) && CUDA_VERSION < 9000
+    ss << R"(
+#define WARP_SIZE 32
+#define __ballot_sync(mask, predicate) __ballot(predicate)
+#define __shfl_down_sync(mask, val, delta, width) __shfl_down(val, delta, width)
+#define __shfl_xor_sync(mask, val, laneMask, width) __shfl_xor(val, laneMask, width)
+)";
+#endif
+
+    // add modern type definitions
+    ss << "typedef signed char int8_t;\n";
```

Comment:
Don't know if it matters but `using` statements are more modern.
`using int8_t = signed char`

Reply:
Yeah, I would prefer using as well, but it is unfortunately not supported by nvrtc.